test/sqlite/sqllogic_test_logger.cpp

```diff
@@ -1,10 +1,15 @@
 #include "sqllogic_test_logger.hpp"
 #include "duckdb/parser/parser.hpp"
 #include "termcolor.hpp"
+#include "result_helper.hpp"
 #include "sqllogic_test_runner.hpp"
+#include "test_helpers.hpp"
 
 namespace duckdb {
 
+static mutex summary_mutex;
+static vector<string> failures_summary;
```

Comment:
Instead of having this be global statics can we move this to a separate class (`FailureSummary` or similar)?